geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -1322,6 +1328,14 @@ private void createBatchSendBuffer() {
     this.batchFlusher.start();
   }
 
+  public void onIdleCancel() {
```

Comment:
We'll change this to cleanUpOnIdleTaskCancel.  The connection will always be closed when we reach this code.  We debated adding an assert here, but decided against it.